examples/cifar10_cnn_capsule.py

```diff
@@ -0,0 +1,211 @@
+'''Train a simple CNN-Capsule Network on the CIFAR10 small images dataset.
+paper: https://arxiv.org/abs/1710.09829
+
+Without Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and overfitting after 20 epcohs
+
+With Data Augmentation:
+It gets to 75% validation accuracy in 10 epochs,
+and 79% after 15 epochs, and 83% after 30 epcohs.
+In my test, highest validation accuracy is 83.79% after 50 epcohs.
+
+This is a fast Implement, just 20s/epcoh with a gtx 1070 gpu.
+'''
+
+
+# the Capsule Implement is from https://github.com/bojone/Capsule/
+
+from __future__ import print_function
+from keras import backend as K
+from keras.engine.topology import Layer
+from keras.layers import Activation
+from keras import utils
+from keras.datasets import cifar10
+from keras.models import Model
+from keras.layers import *
+from keras.preprocessing.image import ImageDataGenerator
+
+
+# a squashing function. but it has litte difference from the Hinton's paper.
+# it seems that this form of squashing performs better.
+def squash(x, axis=-1):
+    s_squared_norm = K.sum(K.square(x), axis, keepdims=True) + K.epsilon()
+    scale = K.sqrt(s_squared_norm) / (0.5 + s_squared_norm)
+    return scale * x
+
+
+# define our own softmax function instead of K.softmax
+def softmax(x, axis=-1):
+    ex = K.exp(x - K.max(x, axis=axis, keepdims=True))
+    return ex / K.sum(ex, axis=axis, keepdims=True)
+
+
+'''
```

Comment:
Please put this comment on top of your file with the description of the example.